src/System.Threading.AccessControl/tests/EventWaitHandleAclTests.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.AccessControl;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Threading.Tests
+{
+    public class EventWaitHandleAclTests : AclTests
+    {
+        [Fact]
+        public void EventWaitHandle_Create_NullSecurity()
+        {
+            CreateAndVerifyEventWaitHandle(
+                initialState: true,
+                mode: EventResetMode.AutoReset,
+                name: GetRandomName(),
+                expectedSecurity: null,
+                expectedCreatedNew: true).Dispose();
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void EventWaitHandle_Create_NameMultipleNew(string name)
+        {
+            var security = GetBasicEventWaitHandleSecurity();
+
+            using EventWaitHandle handle1 = CreateAndVerifyEventWaitHandle(
+                initialState: true,
+                mode: EventResetMode.AutoReset,
+                name,
+                security,
+                expectedCreatedNew: true);
+
+            using EventWaitHandle handle2 = CreateAndVerifyEventWaitHandle(
+                initialState: true,
+                mode: EventResetMode.AutoReset,
+                name,
+                security,
+                expectedCreatedNew: true);
+        }
+
+        [Fact]
+        public void EventWaitHandle_Create_CreateNewExisting()
+        {
+            string name = GetRandomName();
+            var security = GetBasicEventWaitHandleSecurity();
+
+            using EventWaitHandle handle1 = CreateAndVerifyEventWaitHandle(
+                initialState: true,
+                mode: EventResetMode.AutoReset,
+                name,
+                security,
+                expectedCreatedNew: true);
+
+            using EventWaitHandle handle2 = CreateAndVerifyEventWaitHandle(
+                initialState: true,
+                mode: EventResetMode.AutoReset,
+                name,
+                security,
+                expectedCreatedNew: false);
+        }
+
+        [Fact]
+        public void EventWaitHandle_Create_GlobalPrefixNameNotFound()
+        {
+            Assert.Throws<DirectoryNotFoundException>(() =>
+            {
+                string prefixedName = @"GLOBAL\" + GetRandomName();
```

Comment:
Nit: move this out of the Assert.Throws.  In general you want the Assert.Throws to be as tight around the thing that's going to throw as possible.

Reply:
Makes sense. Fixing.